spec/controllers/matches_controller_spec.rb

```diff
@@ -201,17 +204,33 @@
 
   describe "DELETE #destroy" do
     it "destroys the requested match" do
+      sign_in @admin
       match = Match.create! valid_attributes
       expect {
         delete :destroy, params: { id: match.to_param }
       }.to change(Match, :count).by(-1)
     end
 
+    it "destroys the requested match, but the event still exists" do
+      sign_in @admin
+      match = Match.create! valid_attributes
+      expect {
+        delete :destroy, params: { id: match.to_param }
+      }.to change(Event, :count).by(0)
+    end
+
     it "redirects to the event schedule page" do
+      sign_in @admin
       match = Match.create! valid_attributes
       delete :destroy, params: { id: match.to_param }
       expect(response).to redirect_to(event_schedule_url(match.event))
     end
+
+    it "shouldn't let a normal user delete a match" do
+      match = Match.create! valid_attributes
+      delete :destroy, params: { id: match.to_param }
+      expect(response).not_to be_success
```

Comment:
also expect that the match count didn't change i.e. prevent that a access denied message is displayed but the match is deleted anyways

Reply:
Shouldn't this be avoided by cancancan anyways? The delete method won't be executed, because the ability file prohibits this